starfish/test/full_pipelines/api/test_iss_api.py

```diff
@@ -1,43 +1,122 @@
+        dtype=np.uint16
+    )
+    assert np.array_equal(
+        expected_registration_values,
+        registered_primary_image.numpy_array[2, 2, 0, 40:50, 40:50]
+    )
+
+    # filter the data
+    filt = WhiteTophat(masking_radius=15)
+    filtered_dots, filtered_nuclei, filtered_primary_image = [
+        filt.run(img, in_place=False) for img in (dots, nuclei, registered_primary_image)
+    ]
+
+    # assert on a random part of the filtered image
+    expected_filtered_values = np.array(
+        [[63, 160, 22, 67, 0, 0, 0, 0, 183, 272],
+         [481, 529, 150, 64, 45, 114, 0, 0, 0, 0],
+         [1044, 194, 298, 306, 0, 0, 42, 113, 0, 0],
+         [2008, 1059, 533, 148, 322, 95, 124, 0, 0, 18],
+         [2442, 1557, 1048, 784, 857, 123, 162, 107, 65, 0],
+         [2715, 2146, 1739, 1397, 1006, 545, 370, 0, 172, 0],
+         [2868, 2771, 2434, 2426, 1641, 1002, 671, 191, 109, 0],
+         [2686, 3078, 2984, 2601, 2092, 1439, 882, 531, 218, 0],
+         [1992, 2698, 3225, 2832, 2369, 1718, 1478, 1088, 385, 97],
+         [1880, 2122, 3027, 3178, 2398, 1652, 1522, 1375, 724, 285]],
+        dtype=np.uint16
+    )
+    assert np.array_equal(
+        expected_filtered_values,
+        filtered_primary_image.numpy_array[1, 1, 0, 40:50, 40:50]
+    )
 
-    min_sigma = 1.5
-    max_sigma = 5
-    num_sigma = 10
-    threshold = 1e-4
+    # call spots
     gsd = GaussianSpotDetector(
-        min_sigma=min_sigma,
-        max_sigma=max_sigma,
-        num_sigma=num_sigma,
-        threshold=threshold,
-        measurement_type='max',
+        min_sigma=1,
+        max_sigma=10,
+        num_sigma=30,
+        threshold=0.01,
+        measurement_type='mean',
     )
-    blobs_image = dots.numpy_array.reshape(1, *dots_data.shape)
-    intensities = gsd.run(data_stack=image, blobs_image=blobs_image)
-    assert intensities.shape[0] == 5
+    with warnings.catch_warnings():
+        warnings.simplefilter("ignore")
+
+        # blobs = dots; define the spots in the dots image, but then find them again in the stack.
+        blobs_image = dots.max_proj(Indices.ROUND, Indices.Z)
+        intensities = gsd.run(primary_image, blobs_image=blobs_image)
+
+    # assert that the number of spots detected is 99
+    assert intensities.sizes[Features.AXIS] == 99
+
+    # decode
+    codebook_json = (
+        'https://dmf0bdeheu4zf.cloudfront.net/'
+        'TEST/20180821/ISS_TEST/fov_001/codebook.json'
+    )
+    codebook = Codebook.from_json(codebook_json)
+    decoded_intensities = codebook.decode_per_round_max(intensities)
+
+    # decoding identifies 4 genes, each with 1 count
+    genes, gene_counts = np.unique(decoded_intensities['target'], return_counts=True)
+    assert np.array_equal(genes, np.array(['CD68', 'GUS', 'None', 'ST-3', 'VEGF']))
+    assert np.array_equal(gene_counts, [1, 1, 95, 1, 1])
+
+    # segment
+    # TODO ambrosejcarr: do these need to be adjusted for the image size?
```

Comment:
seems unlikely...?

Reply:
Yea, this is actually a question that @dganguli could probably answer pretty quickly -- the watershed in this case is missing a really (visually) obvious second cell, so I was wondering if the parameterization is sensitive.